buildSrc/src/test/kotlin/com/google/firebase/gradle/plugins/PublishingPluginTests.kt

```diff
@@ -137,16 +138,21 @@ licenses {
       )
   }
 
+  private inline fun <reified T : Throwable> buildShouldThrow(crossinline block: () -> Unit) =
+    Assert.assertThrows(UnexpectedBuildFailure::class.java) { block() }
+      .also { assertThat(it).isInstanceOf(T::class.java) } as T
+
   @Test
   fun `Publish with unreleased dependency`() {
     val project1 = Project(name = "childProject1", version = "1.0")
     val project2 =
       Project(name = "childProject2", version = "0.9", projectDependencies = setOf(project1))
 
     subprojectsDefined(project1, project2)
-    val exception =
-      Assert.assertThrows(UnexpectedBuildFailure::class.java) { publish(Mode.RELEASE, project2) }
-    assertThat(exception.message).contains("Failed to release com.example:childProject2")
+    val result = publishAndFail(RELEASE, project2)
+    assertThat(result.task(":checkHeadDependencies")?.outcome).isEqualTo(TaskOutcome.FAILED)
+    // also, if our integ tests fail, we need to add multidex back to installations and add a
+    // removal in our pom transformation for it
```

Comment:
wouldn't this be something we can test now and based on that either leave it out or put it back in?

Reply:
that comment was more for me, I talk to future me a lot in my comments :')

this pr is currently testing exactly that